vsintegration/src/FSharp.Editor/Navigation/NavigateToMetadataService.fs

```diff
@@ -0,0 +1,234 @@
+        match trimmedXml.[0] with
+        | '<' ->    String.Join ("", "<root>", xml, "</root>")
+        | _   ->    let escapedXml = SecurityElement.Escape xml
+                    String.Join ("", "<summary>", escapedXml, "</summary>")
+            
+    let getXmlDocBySignature (fsSymbolUse: FSharpSymbolUse) signature =
+        match fsSymbolUse.Symbol.Assembly.FileName with
+        | Some assemblyName ->
+            match xmlDocCache.TryGetValue assemblyName with
+            | true, xmlIndex -> 
+                match xmlIndex.ParseMemberSignature signature with
+                | _, 0u -> []
+                | _, index ->
+                    match xmlIndex.GetMemberXML index with
+                    | VSConstants.S_OK, xml ->
+                        let processedXml = processXml xml
+                        let xmlDocs = ResizeArray ()
+                        match xmlIndexService.GetMemberDataFromXML processedXml with
+                        | VSConstants.S_OK, memberData ->
+                            match memberData.GetSummaryText () with
+                            | VSConstants.S_OK, xmlSummary -> xmlDocs.Add xmlSummary
+                            | _ -> ()
+                            match memberData.GetParamCount () with
+                            | VSConstants.S_OK, count when count > 0 ->
+                                for i in 0..count-1 do
+                                    match memberData.GetParamTextAt(i) with
+                                    | VSConstants.S_OK, name, text ->
+                                        let xmlDoc = sprintf "%s: %s" name text
+                                        xmlDocs.Add xmlDoc
+                                    | _ -> ()
+                            | _ -> ()
+                            xmlDocs |> Seq.toList
+                        | _ -> []                            
+                    | _ -> []
+            | false, _ -> 
+                match xmlIndexService.CreateXMLMemberIndex assemblyName with
+                | VSConstants.S_OK, xmlIndex ->
+                    match xmlIndex.BuildMemberIndex () with
+                    | VSConstants.S_OK ->
+                        xmlDocCache.Add (assemblyName, xmlIndex)
+                        []
+                    | _ -> []
+                | _ -> []
+        | None -> []
+
+
+    /// Find the range of the target system in the generated signature metadata file
+    let tryFindExactLocation (filePath:string) (source:string) (currentSymbol:FSharpSymbolUse) (options:FSharpProjectOptions) sigCheck = asyncMaybe {
+            let checker = checkerProvider.Checker
+            let! symbolUses =
+                if not sigCheck then asyncMaybe{
+                    let! symbolUses = checker.GetAllUsesOfAllSymbolsInSourceString (options, filePath, source, false)  |> liftAsync
+                    debug "partial assmbly results for symbol - %A in '%s'"currentSymbol.Symbol.FullName filePath
```

Comment:
it's like `printf` only it writes to the debug log. It only executes when the assembly is built in the debug configuration